### PR TITLE
chore: Improve releases

### PR DESCRIPTION
**Pull request by Krizzu**
## Summary

- Updated semantic-release package version
- Added settings to generate CHANGELOG file + auto bump package.json (not happening now)

### CODE REVIEW

File: .releaserc [hunk 1]
```diff
@@ -1,5 +1,19 @@
...
+    "@semantic-release/commit-analyzer",
+    "@semantic-release/release-notes-generator",
+    "@semantic-release/changelog",
+    "@semantic-release/npm",
+    "@semantic-release/github",
+    [
+      "@semantic-release/git",
+      {
+        "assets": ["CHANGELOG.md", "package.json"],
+        "message": "chore(release): ${nextRelease.version} [skip ci]"
```
Comment (tido64) [inline .releaserc:15]: GitHub Actions does not understand `skip ci` by default. We need to manually check for it. You can have a look at how it was done in react-native-test-app: https://github.com/microsoft/react-native-test-app/commit/c20eaaaba93e0af687648def954ff6e65f36194a

File: package.json [hunk 2]
```diff
@@ -92,7 +94,7 @@
     "react-native-web": "~0.12.0",
     "react-native-windows": "^0.63.41",
     "react-test-renderer": "16.13.1",
-    "semantic-release": "^17.2.1"
+    "semantic-release": "^17.4.6"
```
Comment (tido64) [inline package.json:97]: Should we bump to 18?

File: package.json [hunk 3]
```diff
@@ -74,6 +74,8 @@
     "@react-native-community/cli-platform-android": "^4.10.0",
     "@react-native-community/cli-platform-ios": "^4.10.0",
     "@react-native-community/eslint-config": "^3.0.0",
+    "@semantic-release/changelog": "^5.0.1",
+    "@semantic-release/git": "9.0.0",
```
Comment (tido64) [inline package.json:78]:
```suggestion
    "@semantic-release/changelog": "^6.0.1",
    "@semantic-release/git": "^10.0.1",
```

File: .releaserc (hunk 1 above)
Comment (tido64) [reply .releaserc:15]: Never mind. It looks like it was added earlier this year: https://github.blog/changelog/2021-02-08-github-actions-skip-pull-request-and-push-workflows-with-skip-ci/

File: package.json (hunk 2 above)
Comment (Krizzu) [reply package.json:97]: I did not bump to latest due to node version required (14), while our CI images are at 12

File: package.json (hunk 3 above)
Comment (Krizzu) [reply package.json:78]: Same as above, those require node 14

File: .releaserc (hunk 1 above)
Comment (Krizzu) [reply .releaserc:15]: good one, will apply that

File: .circleci/config.yml [hunk 4]
```diff
@@ -40,7 +40,7 @@ default config for android apk builds: &android_defaults
   resource_class: 'medium'
   working_directory: ~/async_storage
   environment:
-    - _JAVA_OPTIONS: '-XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap -Xmx2048m'
```
Comment (Krizzu) [inline .circleci/config.yml:43]: Apparently this was never needed
https://stackoverflow.com/a/52555836/3510245

File: package.json (hunk 2 above)
Comment (tido64) [reply package.json:97]: Should we bump Node to 14 (in a separate PR)?

**Review by tido64 — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (Krizzu) [reply package.json:97]: Yes, that'd require more compatibility checks on CI, so ideally separate PR

**Pull request merged**